### PR TITLE
Display and cache images from available renditions

### PR DESCRIPTION
**Pull request by PeteCoward**
- adds a very simplistic RenditionSelector component used by pre-caching and image display to choose the right rendition
- adds a BaseCard implementation to render new block type from https://github.com/catalpainternational/canoe-backend/pull/5
- adds an ImageWrapper tag to render an image using the RenditionSelector
- moves the Media tag into a better location
- updates the Raw tag to react to the raw `embed` elements by mounting an ImageWrapper

I think we can talk about being more intelligent in the RenditionSelector, and falling back if the rendition is not present.
Also the `riot.component` approach in the Raw tag is something which we could use in more places to make multiple tag mounting more modular ( @sandroroux this is what we tried and failed to do in App a while back I think )

### CODE REVIEW

File: src/js/RenditionSelector.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    return renditionUrl(images[id]);
+}
+
+export function imageUrls(images) {
+    return Object.values(images).map(renditionUrl);
+}
+
+function renditionUrl(renditions) {
+    return `${BACKEND_BASE_URL}/media/${renditions[FILTER_SPEC]}`;
+}
```
Comment (sandroroux) [inline src/js/RenditionSelector.js:17]: These methods should use verbs: `getImagePath`; `getImageUrl`; `getRenditionUrl`.

File: src/riot/Components/BaseCard.riot.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<basecard>
+    <h3 if={ props.card.header }>{ props.card.header }</h3>
+
+    <imagewrapper if={ props.card.image.id }
```
Comment (sandroroux) [inline src/riot/Components/BaseCard.riot.html:4]: @PeteCoward, I know it counters the RiotJS guy's suggestion—place HTML tag attributes in quotes and capitalize tags' nouns:
```html
<ImageWrapper if="{ props.card.image.id }" 
```

The caps follow the codebase's style, and we should fix it all at once.

File: src/riot/Components/ImageWrapper.riot.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<imagewrapper>
+    <img alt="{ props.alt }" 
+        class="richtext-image"
+        crossorigin="anonymous"
+        src="{ getImagePath(props.id) }">
+    </div>
```
Comment (sandroroux) [inline src/riot/Components/ImageWrapper.riot.html:6]: Your closing tag is wrong, I think:
```html
</img>
```

File: src/riot/Components/ImageWrapper.riot.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+    <script>
+        import { imagePath } from "js/RenditionSelector"
+        export default {
+            getImagePath(id) {
+                return imagePath(id);
+            },
```
Comment (sandroroux) [inline src/riot/Components/ImageWrapper.riot.html:13]: Aha! `**get**ImagePath`

File: src/riot/Components/Raw.riot.html [hunk 5]
```diff
@@ -1,8 +1,24 @@
 <raw>
     <script>
+        import * as riot from "riot";
```
Comment (sandroroux) [inline src/riot/Components/Raw.riot.html:3]: Let's avoid importing everything. Can you import `component`?

File: src/riot/Components/Raw.riot.html [hunk 6]
```diff
@@ -1,8 +1,24 @@
...
+                        m.insertAdjacentElement(
+                            "afterend",
+                            document.createElement("div")
+                        ),
+                        {
+                            id:m.getAttribute("id"),
+                            alt:m.getAttribute("alt"),
+                        }
+                    );
+                });
```
Comment (sandroroux) [inline src/riot/Components/Raw.riot.html:21]: this.$$("embed").forEach(m => {
    mountImage(
        embed.insertAdjacentElement(
            "afterend",
            document.createElement("div")
        ),
        {
            id: embed.getAttribute("id"),
            alt: embed.getAttribute("alt"),
        }
    );
});

File: src/riot/Lesson/LessonContent.riot.html [hunk 7]
```diff
@@ -5,6 +5,10 @@
                 if="{props.lessonContent.cards[state.cardIdx].tag === 'raw'}"
                 html="{props.lessonContent.cards[state.cardIdx].content}"
             />
+            <BaseCard
```
Comment (sandroroux) [inline src/riot/Lesson/LessonContent.riot.html:8]: I still don't understand why this is `Base`. What is it the base of? Can the name be more specific?

File: src/js/RenditionSelector.js (hunk 1 above)
Comment (PeteCoward) [reply src/js/RenditionSelector.js:17]: that is so unpythonic, oh wait...__

File: src/riot/Components/BaseCard.riot.html (hunk 2 above)
Comment (PeteCoward) [reply src/riot/Components/BaseCard.riot.html:4]: I will tune my base to yours on the attribute quotes, I know it helps some peoples' crappy editors make things look pretty.

Totally agree on the capitalisation.

File: src/riot/Components/Raw.riot.html (hunk 5 above)
Comment (PeteCoward) [reply src/riot/Components/Raw.riot.html:3]: will try

File: src/riot/Lesson/LessonContent.riot.html (hunk 7 above)
Comment (PeteCoward) [reply src/riot/Lesson/LessonContent.riot.html:8]: I know, I just went with what was in the olgeta issue David made having failed to find a better name

File: src/riot/Components/ImageWrapper.riot.html (hunk 4 above)
Comment (PeteCoward) [reply src/riot/Components/ImageWrapper.riot.html:13]: My fix will give with one hand and take away with the other. I know we use the pattern of same name accessors in other places, but I think it is confusing.

**Comment by PeteCoward** [comment]
@sandroroux since you merged catalpainternational/canoe-backend#5, we need to merge this too!

**Pull request merged**